packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -569,16 +569,21 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
                 const response = await fetchSnapshot(this.snapshotUrl!, storageToken, id, this.fetchFullSnapshot, this.logger);
                 const odspSnapshot: IOdspSnapshot = response.content;
-                // OdspSnapshot contain "trees" when the request is made for latest or the root of the tree, for all other cases it will contain "tree" which is the fetched tree with the id
+                let treeId = "";
                 if (odspSnapshot) {
                     if (odspSnapshot.trees) {
                         this.initTreesCache(odspSnapshot.trees);
+                        if (odspSnapshot.trees.length > 0) {
+                            treeId = odspSnapshot.trees[0].id ?? "";
+                        }
                     }
                     if (odspSnapshot.blobs) {
                         this.initBlobsCache(odspSnapshot.blobs);
                     }
                 }
-                return this.treesCache.get(id);
+                // If the version id doesn't match with the id of the tree, then use the id of first tree which in that case
```

Comment:
I know it's port of prior changes, but some suggestions:
1. I'd not use treeId = "" above, but rather use undefined and not fetch from this.treesCache with empty string. Who knows, maybe in the future there will be some special entry with "" name.
2. I think this change says that there are two set of related IDs. Can we name them properly? I.e. id - what this ID is for? Just looking at the treatment, I'm not even sure that right API is called in this workflow. I.e. this is versionId, not treeID. Can this API be called with either version ID and tree ID?

Reply:
Made treeId to be string or undefined. After talking to marcelo, we reached the conclusion that we should always fetch from the cache with the id of tree.